nextflow_schema.json

```
@@ -41,47 +88,224 @@
                     "type": "string",
-                    "format": "file-path",
-                    "mimetype": "text/plain",
-                    "pattern": "^\\S+\\.fn?a(sta)?(\\.gz)?$",
-                    "description": "Path to FASTA genome file.",
-                    "help_text": "This parameter is *mandatory* if `--genome` is not specified. If you don't have a BWA index available this will be generated for you automatically. Combine with `--save_reference` to save BWA index for future runs.",
-                    "fa_icon": "far fa-file-code"
+                    "description": "Instructs Trim Galore to remove bp from the 5' end of read 2 (or single-end reads).",
+                    "fa_icon": "fas fa-hand-scissors"
+                },
+                "three_prime_clip_r1": {
+                    "type": "string",
+                    "description": "Instructs Trim Galore to remove bp from the 3' end of read 1 AFTER adapter/quality trimming has been performed.",
+                    "fa_icon": "far fa-hand-scissors"
+                },
+                "three_prime_clip_r2": {
+                    "type": "string",
+                    "description": "Instructs Trim Galore to remove bp from the 3' end of read 2 AFTER adapter/quality trimming has been performed.",
+                    "fa_icon": "far fa-hand-scissors"
+                },
+                "trim_reads": {
+                    "type": "string",
+                    "description": "Instructs Trim Galore to apply the --nextseq=X option, to trim based on quality after removing poly-G tails.",
+                    "fa_icon": "fab fa-cuttlefish"
+                },
+                "save_trimmed": {
+                    "type": "boolean",
+                    "description": "Save the trimmed FastQ files in the results directory.",
+                    "fa_icon": "fab fa-cuttlefish"
+                },
+                "skip_trimming": {
+                    "type": "boolean",
+                    "description": "Skip the adapter trimming step.",
+                    "fa_icon": "fas fa-forward"
+                }
+            },
+            "fa_icon": "fas fa-cut"
+        },
+        "BBtools_options": {
+            "title": "BBtools options",
+            "type": "object",
+            "description": "All the options you can use for BBmap or BBduk)",
+            "default": "",
+            "properties": {
+                "sequence_filter": {
+                    "type": "string",
+                    "description": "Instructs BBduk to use a fasta file to filter away sequences before running further analysis.",
+                    "fa_icon": "fas fa-filter"
+                },
+                "bbmap_minid": {
+                    "type": "number",
+                    "default": 0.9,
+                    "description": "Minimal identity for BBmap",
+                    "fa_icon": "fas fa-search-minus"
+                },
+                "save_bam": {
```

I linke the idea that params for specific tools are usually prepended by the tool name, usually thats done here, but not for bbmap in that case. I propose

```suggestion
                "bbmap_save_bam": {
```
and for some more params as well.